tracer/src/Datadog.Tracer.Native/debugger_probes_tracker.cpp

```diff
@@ -75,24 +74,40 @@ void debugger::ProbesMetadataTracker::AddMethodToProbe(const shared::WSTRING& pr
      }
 }
 
-bool debugger::ProbesMetadataTracker::TryGetNextInstrumentedProbeIndex(const shared::WSTRING& probeId, int& probeIndex)
+bool debugger::ProbesMetadataTracker::TryGetNextInstrumentedProbeIndex(const shared::WSTRING& probeId, const ModuleID moduleId, 
+                                                                        const mdMethodDef methodId, int& probeIndex)
 {
      std::lock_guard lock(_probeMetadataMapMutex);
 
-     if (ProbeExists(probeId))
+     std::shared_ptr<ProbeMetadata> probeMetadata;
+     if (TryGetMetadata(probeId, probeMetadata))
      {
-         // Try reuse existing index, if one is available
-         if (!_freeProbeIndices.empty())
+         const auto methodIdentifier = trace::MethodIdentifier(moduleId, methodId);
+
+         // Check if an index previously given
```

Comment:
```suggestion
         // Check if an index was previously assigned for this method
```

Reply:
Done in e3443b9b41459db08459c5cbadf67dd5b0bff45c.